src/nimony/sigmatch.nim

```diff
@@ -1220,6 +1333,8 @@ proc cmpMatches*(a, b: Match): DisambiguationResult =
     else:
       if a.fn.typ.typeKind == ParamsT and b.fn.typ.typeKind == ParamsT:
         result = mutualGenericMatch(a, b)
+        if result == NobodyWins:
+          result = complexDisambiguation(a, b)
```

Comment:
Why is this required? It was more elegant without this logic.

Reply:
Without this overloads for `Ordinal` and `enum` are ambiguous. Since `Ordinal` does not match `enum` and `enum` does not match `Ordinal` with this PR, `mutualGenericMatch` doesn't work. `sumGeneric` calculates the value of `Ordinal` as 0 and `enum` as 1 as in the original compiler.  `==`, `<` etc have overloads for `Ordinal` and `enum` each, and this PR also makes `items` for sets have overloads for `Ordinal` and `enum`.

Reply:
> Since Ordinal does not match enum

How come? Do you mean enum with holes?

Reply:
`T: enum` could be either a holey enum or an ordinal enum, so it does not always satisfy `T: Ordinal`, so it doesn't match. We could make it match but it would fail during instantiation because the overloads are chosen at generic time.

Reply:
What if we introduce a dedicated typeclass `enumgap` to system.nim.

Reply:
The problem of #952 would still exist, it would probably be a big incompatibility that holey enums which are declared as `enum` do not match `enum`

Reply:
An enum without holes should match `enum` and an enum with holes should match the new `enumgap`. Then if both are supported `enum|enumgap` can be used. Ordinal includes `enum` but not `enumgap`. Would this work?

Reply:
> An enum without holes should match  enum`

The problem is that this is an incompatibility. And it doesn't read well, holey enums are declared as `type Foo = enum` same as regular enums. It would work otherwise.

Maybe we could have all 3, `enum` matches any enum, `OrdinalEnum` matches only ordinal enums etc. But still it would be good to be able to disambiguate between `T: enum` and `T: Ordinal` because original Nim already does.

Reply:
The `sumGeneric` code is fine to have in --compat mode but too ugly to go forward with the language in the long run. Esp since the "enum but it's not ordinal" is a problem in Nim 2 too that we never resolved properly. So I would like to have a real solution here.

Reply:
Going to see about splitting `enum` and overloads defined on it then but it might need #931 to work properly (`nifbuilder.addFlags` would break without it but it's unused)